### DISCUSSION TITLE
Access the parts of a Markdown Text

### DISCUSSION THREAD

**Discussion by georgbachmann** [gonzalezreal/swift-markdown-ui#217] [Q&A] [answered]
I have an app, where I would like to display Markdown in all of it's glory. So with all the formatting but also with inline images and so on. But I'd also like my Speech-Synthesizer to be able to speak out the text. But as images in markdown are in general large links, that nobody cares about (spoken), I'd need to be able to access the parts that make up the whole content.

Here's an example:
```
This is some text with an Image
![My image](https://link.to.my.image.jpg)
```

I've seen that `public struct MarkdownContent` contains exactly those parts as `let blocks: [Block]` (https://github.com/gonzalezreal/swift-markdown-ui/blob/a6aeec4cc2edd6af6833bfdcbf8a60db22c263b8/Sources/MarkdownUI/DSL/Blocks/MarkdownContent.swift#L62)

And in the debugger I can now see, that the `MarkdownContent` now has two blocks (for that given example). One text block and one image block. That would be ideal for me to be able to create a string out of it, that I could then speak out.

Or maybe is there any other solution? Or is there a good reason for `enum Block` to not be public?
Should I make an PR making it public?

Thanks so much for you help,
Cheers,
Georg

**Comment by gonzalezreal** [comment]
Hi @georgbachmann,

To keep MarkdownUI public API simple, the `Block` enum should remain an implementation detail. I plan to make the Markdown content available to block styles, as discussed in #201 and in this Twitter thread:

https://twitter.com/twostraws/status/1639263285995356166?s=20

Would that fulfill your use case?

**Reply by georgbachmann** [reply]
Hey @gonzalezreal 
Thank you very much for your reply and help!!! Greatly appreciated!
Hm... I am not a 100% sure. So that would expose it to Swift-UI then?
I'd need it more in my logic. So that in that place, where I prepare my UI, I can also prepare the "speakable" part of the text. And that would not be part of my UI...
Or do I miss-understand the discussion?

**Reply by gonzalezreal** [reply, marked as answer]
Hey @georgbachmann,

It depends on how you'd like the UI to trigger the text-to-speech. With the changes introduced in [MarkdownUI 2.1.0](https://github.com/gonzalezreal/swift-markdown-ui/releases/tag/2.1.0), you can access a plain text representation of any `MarkdownContent` value by using [`renderPlainText()`](https://swiftpackageindex.com/gonzalezreal/swift-markdown-ui/2.1.0/documentation/markdownui/markdowncontent/renderplaintext()).

So, depending on what you want to do, you can construct a `MarkdownContent` value and get its text representation before creating a `Markdown` view, or you can access the `MarkdownContent` value of a specific type of block (i.e., a paragraph) by overriding its style.

**Reply by georgbachmann** [reply]
Ah... cool.. that sounds very good! thx!

**Reply by ChuanqingYang** [reply]
What If I want to access some type of markdown text (eg: H1 )? By using `renderPlainText()` I can only get the full text not what I want.
